### PR TITLE
[FIX] /groups.invite not allow a user to invite even with permission

### LINKED ISSUE #10639: [error-room-not-found] when trying to add user to a private group from API with [add-user-to-any-p-room] permission
### Description:

I'm trying to add a registered user to a private group. I'm posting a request via API (https://rocket.chat/docs/developer-guides/rest-api/groups/invite/) with admin credentials. The admin is not a member of the private group but should be able to add another user with the add-user-to-any-p-room permission. I keep getting 'error-room-not-found' as error type. However the request seems to work when the admin is member of the private group. 

### Server Setup Information:

* Version of Rocket.Chat Server: 0.64.0-develop
* Operating System: macOS 10.13.4 High Sierra
* Deployment Method(snap/docker/tar/etc): 
* Number of Running Instances: 1
* DB Replicaset Oplog: Enabled
* Node Version: v8.11.1
* mongoDB Version: 3.4.10

### Steps to Reproduce:

1. Enable add-user-to-any-p-room permission for user that is sending the request
2. Create second user
3. Set request headers x-auth-token and x-user-id
3. Send POST request to http://localhost:3000/api/v1/groups.invite
Body:
{
	"userId": "75c9mWS83yCTcqn2X", // straight from database
	"roomId": "dnqSmTG52uCzyPm3j" // straight from database
}

Same request works when the admin is member of the private group. 

### Expected behavior:

User is added to the group while admin is still not a member. 

### Actual behavior:

Response with error type "error-room-not-found'

### Relevant logs:
Nothing relevant coming from logs.

Response: 
{
    "success": false,
    "error": "The required \"roomId\" or \"roomName\" param provided does not match any group [error-room-not-found]",
    "errorType": "error-room-not-found"
}

error thrown in Rocket.Chat/packages/rocketchat-api/server/v1/groups.js line 17

### PR DESCRIPTION
**Pull request by MarcosSpessatto**
Closes #10639

### CODE REVIEW

File: packages/rocketchat-api/server/v1/groups.js [hunk 1]
```diff
@@ -348,7 +348,23 @@ RocketChat.API.v1.addRoute('groups.info', { authRequired: true }, {
 
 RocketChat.API.v1.addRoute('groups.invite', { authRequired: true }, {
 	post() {
-		const findResult = findPrivateGroupByIdOrName({ params: this.requestParams(), userId: this.userId });
+		let findResult;
+		const canAddUserToAnyPrivateGroup = RocketChat.authz.hasPermission(this.userId, 'add-user-to-any-p-room');
+		const params = this.requestParams();
+		if (canAddUserToAnyPrivateGroup) {
+			if (params.roomId && params.roomId.trim()) {
+				findResult = RocketChat.models.Subscriptions.findByRoomId(params.roomId).fetch()[0];
```
Comment (Hudell) [inline packages/rocketchat-api/server/v1/groups.js]: Why not find the room directly? Looking for subscriptions will cause the code to fail on empty rooms.

File: packages/rocketchat-lib/server/models/Subscriptions.js [hunk 2]
```diff
@@ -54,6 +54,14 @@ class ModelSubscriptions extends RocketChat.models._Base {
 		return this.findOne(query);
 	}
 
+	findOneByRoomName(roomName) {
```
Comment (Hudell) [inline packages/rocketchat-lib/server/models/Subscriptions.js]: If you use the room directly, this method won't be necessary.

**Comment by Hudell** [comment]
@ggazzo Those methods don't work if you've not joined the room.

File: packages/rocketchat-api/server/v1/groups.js [hunk 3]
```diff
@@ -348,12 +348,27 @@ RocketChat.API.v1.addRoute('groups.info', { authRequired: true }, {
...
+			}
+		} else {
+			findResult = findPrivateGroupByIdOrName({ params, userId: this.userId });
+		}
 
 		const user = this.getUserFromParams();
 
 		Meteor.runAsUser(this.userId, () => {
-			Meteor.call('addUserToRoom', { rid: findResult.rid, username: user.username });
+			Meteor.call('addUserToRoom', { rid: findResult._id, username: user.username });
```
Comment (Hudell) [inline packages/rocketchat-api/server/v1/groups.js]: This gets the id of the object, which is valid when it's a room, but when the user doesn't have the 'add-user-to-any-p-room', the findResult object is still going to be a subscription.

It's better to assign the id to a variable inside the conditional (room._id when the user has the permission, or subscription.rid when the user doesn't have it) and then use the variable here.

**Comment by Hudell** [comment]
The current solution works perfectly. Can you check if there are more endpoints using the `findPrivateGroupByIdOrName` method, where the same problem may be happening?

**Comment by ggazzo** [comment]
I think for this case you should just call the dpp method

**Comment by q16marvin** [comment]
i have still exakt this problem, admin with permission "add-user-to-any-p-room" cant add users to private rooms where admin is not a member, what do i wrong?

**Pull request merged**